benchmark/worker/benchmark_client.go

```diff
@@ -0,0 +1,330 @@
+	stop          chan bool
+	mu            sync.RWMutex
+	lastResetTime time.Time
+	histogram     *stats.Histogram
+}
+
+func startBenchmarkClientWithSetup(setup *testpb.ClientConfig) (*benchmarkClient, error) {
+	var opts []grpc.DialOption
+
+	// Some setup options are ignored:
+	// - client type:
+	//     will always create sync client
+	// - async client threads.
+	// - core list
+	grpclog.Printf(" * client type: %v (ignored, always creates sync client)", setup.ClientType)
+	switch setup.ClientType {
+	case testpb.ClientType_SYNC_CLIENT:
+	case testpb.ClientType_ASYNC_CLIENT:
+	default:
+		return nil, grpc.Errorf(codes.InvalidArgument, "unknow client type: %v", setup.ClientType)
+	}
+	grpclog.Printf(" * async client threads: %v (ignored)", setup.AsyncClientThreads)
+	grpclog.Printf(" * core list: %v (ignored)", setup.CoreList)
+
+	grpclog.Printf(" - security params: %v", setup.SecurityParams)
+	if setup.SecurityParams != nil {
+		creds, err := credentials.NewClientTLSFromFile(Abs(caFile), setup.SecurityParams.ServerHostOverride)
+		if err != nil {
+			return nil, grpc.Errorf(codes.InvalidArgument, "failed to create TLS credentials %v", err)
+		}
+		opts = append(opts, grpc.WithTransportCredentials(creds))
+	} else {
+		opts = append(opts, grpc.WithInsecure())
+	}
+
+	grpclog.Printf(" - core limit: %v", setup.CoreLimit)
+	// Use one cpu core by default
+	numOfCores := 1
+	if setup.CoreLimit > 0 {
+		numOfCores = int(setup.CoreLimit)
+	}
+	runtime.GOMAXPROCS(numOfCores)
+
+	grpclog.Printf(" - payload config: %v", setup.PayloadConfig)
+	var payloadReqSize, payloadRespSize int
+	var payloadType string
+	if setup.PayloadConfig != nil {
+		switch c := setup.PayloadConfig.Payload.(type) {
+		case *testpb.PayloadConfig_BytebufParams:
+			opts = append(opts, grpc.WithCodec(byteBufCodec{}))
+			payloadReqSize = int(c.BytebufParams.ReqSize)
+			payloadRespSize = int(c.BytebufParams.RespSize)
+			payloadType = "bytebuf"
+		case *testpb.PayloadConfig_SimpleParams:
+			payloadReqSize = int(c.SimpleParams.ReqSize)
+			payloadRespSize = int(c.SimpleParams.RespSize)
+			payloadType = "protobuf"
+		case *testpb.PayloadConfig_ComplexParams:
+			return nil, grpc.Errorf(codes.Unimplemented, "unsupported payload config: %v", setup.PayloadConfig)
+		default:
+			return nil, grpc.Errorf(codes.InvalidArgument, "unknow payload config: %v", setup.PayloadConfig)
+		}
+	}
+
+	grpclog.Printf(" - rpcs per chann: %v", setup.OutstandingRpcsPerChannel)
+	grpclog.Printf(" - channel number: %v", setup.ClientChannels)
+
+	rpcCountPerConn, connCount := int(setup.OutstandingRpcsPerChannel), int(setup.ClientChannels)
+
+	grpclog.Printf(" - load params: %v", setup.LoadParams)
+	var dist *int
+	switch lp := setup.LoadParams.Load.(type) {
+	case *testpb.LoadParams_ClosedLoop:
+	case *testpb.LoadParams_Poisson:
+		grpclog.Printf("   - %v", lp.Poisson)
+		return nil, grpc.Errorf(codes.Unimplemented, "unsupported load params: %v", setup.LoadParams)
+		// TODO poisson
```

Comment:
remove. and add a comment to add more dist above the line:
switch lp := setup.LoadParams.Load.(type).